Button 컴포넌트 + ModalDialog 컴포넌트 구현

joowon-jang opened the pull request:
## PR 유형

- [x] Feat: 새로운 기능 추가
- [ ] BugFix: 버그 수정
- [x] Design: CSS 등 사용자 UI 디자인 변경
- [ ] Style: 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] Refactor: 코드 리팩토링
- [ ] Comment: 주석 추가 및 수정
- [ ] Docs: 문서 수정
- [ ] Test: 테스트 추가, 테스트 리팩토링
- [ ] Chore: 빌드 부분 혹은 패키지 매니저 수정
- [ ] Rename: 파일 혹은 폴더명 수정
- [ ] Remove: 파일 혹은 폴더 삭제

## 🔧 반영 브랜치

<!-- ex) feature/login -> develop -->
feature/ModalDialog -> develop

## 💬 Description

<!-- 리뷰하는 사람이 봐야할 내용을 작성해주세요. -->
<!-- 구현 한 기능에 대해 작성해 주세요. -->
- Button 컴포넌트는 사용성을 고려해서 width 100%로 했습니다. (Button 사용할 때, wrapper나 container크기를 맞춰서 사용)
- ModalDialog 컴포넌트의 버튼 위쪽 컨텐츠는 임의로 넣었습니다. (컨텐츠 style은 컴포넌트 사용할 때 설정해서 사용)
- useBodyScrollLock 훅은 모달 창이 열린 상태에서는 바깥쪽 스크롤이 되지 않도록 방지하는 기능입니다.

## 📷 Screenshots

<!--스크린샷으로 보여줄 수 있는 이미지가 있다면 첨부해주세요!-->
<!-- 스크린샷은 선택사항입니다. -->

<img width="200" alt="스크린샷 2025-01-23 오후 12 22 14" src="https://github.com/user-attachments/assets/117ab86c-904b-4104-a60f-5d76e3f56044" />

<img width="200" alt="스크린샷 2025-01-23 오후 12 22 34" src="https://github.com/user-attachments/assets/bddf6783-7e75-43e9-9bf7-8ac10fe39d49" />

<img width="200" alt="스크린샷 2025-01-23 오후 12 23 01" src="https://github.com/user-attachments/assets/99599144-1547-4277-940f-65e5591a3556" />

<!--
맨 마지막 줄에 아래 이슈
Fixes: 이슈 수정중 (아직 해결되지 않은 경우)
Resolves: 이슈를 해결했을 때 사용
Ref: 참고할 이슈가 있을 때 사용
Related to: 해당 커밋에 관련된 이슈번호 (아직 해결되지 않은 경우)

-여러 이슈를 입력시 comma(,) 단위로 구분해주세요
ex) Close #10, Resolves #123 -->
Resolves #4

joowon-jang committed:
feat: Button 컴포넌트 구현

joowon-jang committed:
feat: useBodyScrollLock 커스텀 훅 작성

joowon-jang committed:
feat: ModalDialog 컴포넌트 구현

joowon-jang committed:
design: 전역 스타일 수정

- button, input, select의 border 제거
- 전역 색상 변수 black 추가

yezzero commented on components/Button.tsx:
```diff
@@ -0,0 +1,59 @@
+/* ---------------------------------- type ---------------------------------- */
+type ButtonProps = {
+  children?: string;
+  type: 'long' | 'short';
```
type 말고 isLong으로 불리언 값으로 하는 거 어ㄸㅐ요?!? type이라는 이름이 원래 존재하는 type과 헷가렬욤

joowon-jang committed:
modify: Butotn 컴포넌트의 type prop을 isLong으로 변경

joowon-jang committed:
refactor: ModalDialog 컴포넌트 수정

- 버튼이 없는 타입 추가
- type -> buttonCount로 prop이름 수정
- 버튼이 없는 모달 창의 경우 추가에 따라 onClose prop을 옵셔널로 변경

joowon-jang committed:
style: font-weight 속성 bold로 통일

joowon-jang committed:
design: ModalDialog 컴포넌트 z-index 추가

- Header, Navbar도 fixed이기 때문에 z-index를 설정해서 모달이 덮을 수 있도록 함

joowon-jang committed:
style: Button 컴포넌트의 styled-components props 이름 변경

- $표시 붙이는 것으로 통일

Taemin1 reviewed (approved):
확인했습니다

yezzero reviewed (approved):
굿좝
